api/api.py

```diff
@@ -441,6 +493,24 @@ async def get_ci_badge_get(repo: str, branch: str, workflow:str):
 
 # Helper functions, not directly callable through routes
 
+def escape_dict(dictionary):
+    for key, value in dictionary.items():
+        if isinstance(value, str):
+            dictionary[key] = escape(value, quote=False)
+            continue
+        if isinstance(value, dict):
+            escape_dict(value)
+            continue
+        if isinstance(value, list):
+            for item in value:
+                if isinstance(item, str):
+                    dictionary[key] = escape(item, quote=False)
+                    continue
+                if isinstance(item, dict):
+                    escape_dict(item)
+
+    return dictionary
+
```

Comment:
I don't think this will work properly as you sometimes just call `escape_dict` without assigning it?

```
def escape_dict(dictionary):
    for key, value in dictionary.items():
        if isinstance(value, str):
            dictionary[key] = escape(value, quote=False)
        elif isinstance(value, dict):
            dictionary[key] = escape_dict(value)
        elif isinstance(value, list):
            dictionary[key] = [escape_dict(item) if isinstance(item, dict) else escape(item, quote=False) if isinstance(item, str) else item for item in value]
    return dictionary
```

Something like this might work better. I haven' tested my code just as a reference